lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescaleset.py

```diff
@@ -601,6 +665,10 @@ def exec_module(self, **kwargs):
             elif self.image:
                 self.fail("parameter error: expecting image to be a string or dict not {0}".format(type(self.image).__name__))
 
+            if self.plan:
+                if not self.plan.get('name') or not self.plan.get('product') or not self.plan.get('publisher'):
+                    self.fail("parameter error: plan must include name, product, and publisher")
```

Comment:
This won't be necessary anymore if you add the suboptions to the argument spec. Ansible will do this validation for you then (and also make sure the options have the correct types).

Reply:
Code block removed because I did add the suboptions as suggested. Change will be in my next push.